### DISCUSSION TITLE
`Cache-Control` header ignored when set inside a page endpoint

### DISCUSSION THREAD

**Discussion by matiadev** [sveltejs/kit#4747] [Q&A] [answered]
Hey! 👋

I'm stumped why `Cache-Control` headers are being ignored when set inside a page endpoint. It works when set inside `load` as expected but for some reason it's ignored when set inside a page endpoint.

```html
<!-- index.svelte -->
<script context="module">
  export async function load() {
    return {
      cache: { maxage: 60 } // ✅
    }
  }
</script>
```
```js
// index.js
export async function get() {
  return {
    headers: { 'Cache-Control': 'max-age=60' } // ❌
  }
}
```

**Comment by matiadev** [comment, marked as answer]
From what it seems page endpoints add the `Cache-Control` header to `__data.json` which makes sense but I assumed it would also work for the document. Doesn't that mean the user won't get the cached page when they reload?

**Reply by robots4life** [reply]
> Doesn't that mean the user won't get the cached page when they reload?

@mattcroat 

Does this mean the user **will** get the cached page upon reload ?

How can a user get a cached page when the `Cache-Control` header is set in the page endpoint ?

Were you able to find out and could you share your finding if this is possible at all ?

**Reply by matiadev** [reply]
> Does this mean the user will get the cached page upon reload ?

As you pointed out it doesn't because it's set for the page endpoint.

> Were you able to find out and could you share your finding if this is possible at all ?

Yes! After reading https://github.com/sveltejs/kit/issues/3532 and [loading](https://kit.svelte.dev/docs/loading#input-props) I realized you can pass `props` to the `load` function and set the cache for the document that way. I opened a new discussion at https://github.com/sveltejs/kit/discussions/4814 where you can see an example.

Hope it helps! 😄
